geopandas/base.py

```diff
@@ -495,14 +495,24 @@ def buffer(self, distance, resolution=16, **kwargs):
 
         Parameters
         ----------
-        distance : float
-            The radius of the buffer.
+        distance : float, np.array, pd.Series
+            The radius of the buffer. If np.array or pd.Series are used
+            then it must have same length as the GeoSeries.
         resolution: float
             Optional, the resolution of the buffer around each vertex.
         """
+        if isinstance(distance, (np.ndarray, pd.Series)):
+            if distance.shape[0] != self.shape[0]:
```

Comment:
You could consider using `if len(distance.index) != len(self.shape.index):` perhaps? Similarly for the change below. This is marginally better but really that expensive. See. #787

Reply:
@jdmcbr any opinion on this ?

Reply:
Yep, fine to change to use len()

Reply:
In addition, I think we should check in case of Series also that the index is equal (because if it is not equal, people might expect automatic alignment)